sharktank/sharktank/types/sharding.py

```diff
@@ -93,3 +93,58 @@ def theta_sharding(self) -> ThetaSharding:
                 "bias": Split(shard_count=self.shard_count, shard_dim=0),
             }
         )
+
+
+class LinearReplicatedInputSplitWeightAndBiasSharding(ThetaLayerSharding):
+    def __init__(self, shard_count: int, weight_and_bias_spit_dim: int = 0):
+        super(Sharding).__init__()
+        self.shard_count = shard_count
+        self.weight_and_bias_spit_dim = weight_and_bias_spit_dim
+
+    def theta_sharding(self) -> ThetaSharding:
+        return ThetaSharding(
+            {
+                "premul_input": Replicated(shard_count=self.shard_count),
+                "weight": Split(
+                    shard_count=self.shard_count,
+                    shard_dim=self.weight_and_bias_spit_dim,
+                ),
+                "bias": Split(
+                    shard_count=self.shard_count,
+                    shard_dim=self.weight_and_bias_spit_dim,
+                ),
+            }
+        )
+
+
+class ResnetBlock2DSplitOutputChannelsSharding(ThetaLayerSharding):
```

Comment:
Ok to have this here for now, but it should probably be with the resnet layer (which is not a "general" resnet but specific to unet). Fine for now.

Reply:
Moved them to punet.